lib/ecto/query/builder/select.ex

```diff
@@ -58,6 +58,16 @@ defmodule Ecto.Query.Builder.Select do
     {{:{}, [], [:%{}, [], [{:{}, [], [:|, [], [data, pairs]]}]]}, params_take}
   end
 
+  # filter
```

Comment:
Let's move this to the builder. The reason is that this construct is also allowed on UPDATEs according to Postgres documentation. So let's allow everything on the Ecto side. If the user writes an invalid query, they can raise.

Reply:
:+1: